src/sentry/utils/safe.py

```diff
@@ -141,7 +141,7 @@ def get_path(data, *path, **kwargs):
     for p in path:
         if isinstance(data, collections.Mapping) and p in data:
             data = data[p]
-        elif isinstance(data, (list, tuple)) and -len(data) <= p < len(data):
+        elif isinstance(data, (list, tuple)) and isinstance(p, int) and -len(data) <= p < len(data):
```

Comment:
The tests reflected that string numbers should NOT work, I've made this explicit.

Reply:
Did it work previously in py2?

Reply:
Hmm, the 4th assert in here implies it shouldn't

https://github.com/getsentry/sentry/blob/69e8ccd94246d6674c7ff9eec7efb38d61d50955/tests/sentry/utils/test_safe.py#L129-L135